mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocFinder.java

```diff
@@ -134,6 +136,9 @@ public long find(CRUDOperationContext ctx,
             }            
             return numMatched;
         } finally {
+            if(cursor!=null&&!cursorInUse) {
```

Comment:
line 125: cursor is returned in CursorStream. In that case it should not be closed,